builtin/logical/database/path_roles.go

```diff
@@ -2,263 +2,606 @@ package database
-			"max_ttl":               role.MaxTTL.Seconds(),
-		}
-		if len(role.Statements.Creation) == 0 {
-			data["creation_statements"] = []string{}
+		db.RLock()
+		defer db.RUnlock()
+
+		if err := db.RevokeUser(ctx, role.Statements, role.StaticAccount.Username); err != nil {
+			b.CloseIfShutdown(db, err)
+			return nil, err
 		}
-		if len(role.Statements.Revocation) == 0 {
-			data["revocation_statements"] = []string{}
+	}
+
+	err = req.Storage.Delete(ctx, databaseStaticRolePath+name)
+	if err != nil {
+		return nil, err
+	}
+
+	if b.credRotationQueue != nil {
+		b.credRotationQueue.PopByKey(name)
+	}
+
+	return nil, nil
+}
+
+func (b *databaseBackend) pathStaticRoleRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	role, err := b.StaticRole(ctx, req.Storage, d.Get("name").(string))
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return nil, nil
+	}
+	data := pathRoleReadCommon(role)
+	if role.StaticAccount != nil {
+		data["username"] = role.StaticAccount.Username
+		data["rotation_period"] = role.StaticAccount.RotationPeriod.Seconds()
+		if !role.StaticAccount.LastVaultRotation.IsZero() {
+			data["last_vault_rotation"] = role.StaticAccount.LastVaultRotation
 		}
-		if len(role.Statements.Rollback) == 0 {
-			data["rollback_statements"] = []string{}
+	}
+
+	return &logical.Response{
+		Data: data,
+	}, nil
+}
+
+func (b *databaseBackend) pathRoleRead(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	role, err := b.Role(ctx, req.Storage, d.Get("name").(string))
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return nil, nil
+	}
+
+	return &logical.Response{
+		Data: pathRoleReadCommon(role),
+	}, nil
+}
+
+func pathRoleReadCommon(role *roleEntry) map[string]interface{} {
+	data := map[string]interface{}{
+		"db_name":               role.DBName,
+		"creation_statements":   role.Statements.Creation,
+		"revocation_statements": role.Statements.Revocation,
+		"rollback_statements":   role.Statements.Rollback,
+		"renew_statements":      role.Statements.Renewal,
+		"rotation_statements":   role.Statements.Rotation,
+		"default_ttl":           role.DefaultTTL.Seconds(),
+		"max_ttl":               role.MaxTTL.Seconds(),
+	}
+	if len(role.Statements.Creation) == 0 {
+		data["creation_statements"] = []string{}
+	}
+	if len(role.Statements.Revocation) == 0 {
+		data["revocation_statements"] = []string{}
+	}
+	if len(role.Statements.Rollback) == 0 {
+		data["rollback_statements"] = []string{}
+	}
+	if len(role.Statements.Renewal) == 0 {
+		data["renew_statements"] = []string{}
+	}
+	if len(role.Statements.Rotation) == 0 {
+		data["rotation_statements"] = []string{}
+	}
+	return data
+}
+
+func (b *databaseBackend) pathRoleList(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	path := databaseRolePath
+	if req.Path == "static-roles/" {
```

Comment:
Should this be `strings.HasPrefix(req.Path, "static-roles")`? Wondering if the `/` will always be present (though maybe the framework does just that?)

Reply:
I my limited investigation I found `req.Path` always included the trailing `/`, so I assumed the framework enforced/included that. Having said that, I like your suggestion, and applied it in 4081d2780